experiments/benchmarks/richards.jl

```diff
@@ -0,0 +1,372 @@
+
+    # Below, the preprocess_func argument is used to
+    # 1. Convert precipitation to be negative (as it is downwards)
+    # 2. Convert accumulations over an hour to a rate per second
+    ref_time = DateTime(2021)
+    t_start = 0.0
+    # Precipitation:
+    precip = TimeVaryingInput(
+        joinpath(era5_artifact_path, "era5_2021_0.9x1.25.nc"),
+        "tp",
+        surface_space;
+        reference_date = ref_time,
+        t_start,
+        regridder_type,
+        file_reader_kwargs = (; preprocess_func = (data) -> -data / 3600,),
+    )
+    atmos = ClimaLand.PrescribedPrecipitation{FT, typeof(precip)}(precip)
+    bottom_bc = ClimaLand.Soil.WaterFluxBC((p, t) -> 0.0)
+    bc = (;
+        top = ClimaLand.Soil.RichardsAtmosDrivenFluxBC(atmos, runoff_model),
+        bottom = bottom_bc,
+    )
+    model = ClimaLand.Soil.RichardsModel{FT}(;
+        parameters = soil_params,
+        domain = domain,
+        boundary_conditions = bc,
+        sources = (),
+        lateral_flow = false,
+    )
+
+    Y, p, t = initialize(model)
+    z = ClimaCore.Fields.coordinate_field(domain.space.subsurface).z
+    lat = ClimaCore.Fields.coordinate_field(domain.space.subsurface).lat
+    function hydrostatic_profile(
+        lat::FT,
+        z::FT,
+        ν::FT,
+        θ_r::FT,
+        α::FT,
+        n::FT,
+        S_s::FT,
+        fmax,
+    ) where {FT}
+        m = 1 - 1 / n
+        zmin = FT(-50.0)
+        zmax = FT(0.0)
+
+        z_∇ = FT(zmin / 5.0 + (zmax - zmin) / 2.5 * (fmax - 0.35) / 0.7)
+        if z > z_∇
+            S = FT((FT(1) + (α * (z - z_∇))^n)^(-m))
+            ϑ_l = S * (ν - θ_r) + θ_r
+        else
+            ϑ_l = -S_s * (z - z_∇) + ν
+        end
+        return FT(ϑ_l)
+    end
+
+    # Set initial state values
+    Y.soil.ϑ_l .= hydrostatic_profile.(lat, z, ν, θ_r, vg_α, vg_n, S_s, f_max)
+    @. Y.soil.ϑ_l = oceans_to_zero(Y.soil.ϑ_l, mask)
+
+    # Create model update functions
+    set_initial_cache! = make_set_initial_cache(model)
+    exp_tendency! = make_exp_tendency(model)
+    imp_tendency! = make_imp_tendency(model)
+    tendency_jacobian! = make_tendency_jacobian(model)
+
+    set_initial_cache!(p, Y, t0)
+
+    # set up jacobian info
+    jac_kwargs = (;
+        jac_prototype = ClimaLand.Soil.ImplicitEquationJacobian(Y),
+        Wfact = tendency_jacobian!,
+    )
+
+    prob = SciMLBase.ODEProblem(
+        CTS.ClimaODEFunction(
+            T_exp! = exp_tendency!,
+            T_imp! = SciMLBase.ODEFunction(imp_tendency!; jac_kwargs...),
+            dss! = ClimaLand.dss!,
+        ),
+        Y,
+        (t0, tf),
+        p,
+    )
+    save_every = 100
+    saveat = Array(t0:(save_every * Δt):tf)
+    sv = (;
+        t = Array{Float64}(undef, length(saveat)),
+        saveval = Array{NamedTuple}(undef, length(saveat)),
+    )
+    saving_cb = ClimaLand.NonInterpSavingCallback(sv, saveat)
+    updateat = Array(t0:Δt:tf)
```

Comment:
We may update less frequently but I think this is OK for now, unless updating takes a lot of time